package/MDAnalysis/core/topologyattrs.py

```diff
@@ -516,24 +516,20 @@ def center_of_mass(group, **kwargs):
             If ``True``, move all atoms within the primary unit cell before
             calculation. [``False``]
 
-        .. note::
+        Returns
+        -------
+        center : ndarray
+            center of group given masses as weights
+
+        Notes
+        -----
             The :class:`MDAnalysis.core.flags` flag *use_pbc* when set to
             ``True`` allows the *pbc* flag to be used by default.
 
         .. versionchanged:: 0.8 Added `pbc` parameter
         """
-        atomgroup = group.atoms
-
-        pbc = kwargs.pop('pbc', flags['use_pbc'])
-        masses = atomgroup.masses
-
-        if pbc:
-            positions = atomgroup.pack_into_box(inplace=False)
-        else:
-            positions = atomgroup.positions
-
-        return np.sum(positions * masses[:, np.newaxis],
-                      axis=0) / masses.sum()
+        return group.atoms.center(weights=group.atoms.masses,
```

Comment:
With the following small benchmark

```python
from MDAnalysisTests.datafiles import PSF, DCD
import MDAnalysis as mda

u = mda.Universe(PSF, DCD)
u.transfer_to_memory()

%timeit for ts in u.trajectory: u.atoms.center_of_mass()
```

I get 22.2 ms for `develop` and 24.1 ms for this branch. That is about 8% performance penalty. That can be reduced to the same old timing if I use `(weights[:, np.newaxis] * xyz).sum(0)` in `center` instead of `np.average`. I would also be fine with that. We lose a little bit of error reporting and will only be able to use weights for each atom and not for each individual coordinate (weighting only in xy direction movement might be interesting for membrane simulations)